### PR TITLE
Add carousal Slider components

### PR DESCRIPTION
**Pull request by vikas-shrma**
-try with dummy pages
-use static image link as  data source to disaply images

### CODE REVIEW

File: lib/main.dart [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        // application has a blue toolbar. Then, without quitting the app, try
+        // changing the primarySwatch below to Colors.green and then invoke
+        // "hot reload" (press "r" in the console where you ran "flutter run",
+        // or simply save your changes to "hot reload" in a Flutter IDE).
+        // Notice that the counter didn't reset back to zero; the application
+        // is not restarted.
+        primarySwatch: Colors.blue,
+      ),
+       home: const AppCarousel(),
+       //home: const MyHomePage(title: 'AnxietyGo'),
```
Comment (markhallak) [inline lib/main.dart]: Revert the changes on line 31

the home page should be MyHomePage(title: 'AnxietyGo')

File: pubspec.yaml [hunk 2]
```diff
@@ -36,6 +36,7 @@ dependencies:
   # The following adds the Cupertino Icons font to your application.
   # Use with the CupertinoIcons class for iOS style icons.
   cupertino_icons: ^1.0.2
+  carousel_slider: ^4.2.1
```
Comment (markhallak) [inline pubspec.yaml]: Remove the ^ so we can use a specific version and decrease the build time.

File: lib/components/carousal_slider.dart [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  final double? width;
+  final List<String>? imageList;
+  @override
+  State<StatefulWidget> createState() => _AppCarousel();
+}
+
+class _AppCarousel extends State<AppCarousel> {
+  CarouselController carouselController = CarouselController();
+
+  final imageList = [
```
Comment (markhallak) [inline lib/components/carousal_slider.dart]: Figure out a way to send the data from another class or page because in the future we will be getting the data from a firebase database and we will pass the data to the carousel after it has been built. Take care of some edge cases such as what if the data returned is empty.

Also, we might need a static image to display in case the data returned from the database is empty.

**Review by markhallak — CHANGES_REQUESTED** [review]

**Comment by markhallak** [comment]
Hey @vikas-shrma 

Any updates? Are you facing any problems?

And are you working on it with @Prince9294 ?

**Comment by markhallak** [comment]
@vikas-shrma @Prince9294 

Whoever is working on this PR, please make sure there are no typos because there is a recent conflict in this branch in the "pubspec.lock" file

image

**Comment by markhallak** [comment]
Seems like it is solved now. We still need to make sure that the project is running normally.

**Comment by vikas-shrma** [comment]
You have to create a .gitignore file and add pubspec.lock file inside itBecause it will update everytime when different users commit code. Its depends upon system to system Sent from Android deviceOn 26-Feb-2023 2:52 am, markhallak ***@***.***> wrote:
@vikas-shrma @Prince9294
Whoever is working on this PR, please make sure there are no typos because there was a recent conflict in this branch in the "pubspec.lock" file

**Comment by vikas-shrma** [comment]
@markhallak any other changes required. I tested it with dummy code and its working in various conditions

**Comment by markhallak** [comment]
@vikas-shrma 

You're right. I forgot about it.

Regarding the PR, everything seems fine. I will merge it now. Thanks for the help!
